processor/src/test/java/org/mapstruct/ap/test/overloading/OverloadingTest.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.overloading;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+import java.util.Date;
+
+@WithClasses({
+        SourceTargetMapper.class,
+        SourceTargetMapperPropertyBean.class,
+        Source.class,
+        Target.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+public class OverloadingTest {
+
+    @Test
+    public void testShouldGenerateCorrectMapperImplementation() {
+        Source source = new Source( new Date() );
+        Target target = SourceTargetMapper.INSTANCE.sourceToTarget( source );
+        Assert.assertTrue( target.getUpdatedOn() > 0 );
```

Comment:
@agudian is right for the fixture. Just to make sure that we don't miss this in the future, maybe the `setUpdateOn(long updateOn)` in the `Target` should throw an exception. That method should not be called.